subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c

```diff
@@ -302,7 +328,8 @@ int img_mgmt_write_pending(int slot, bool permanent)
 {
 	int rc;
 
-	if (slot != 1 && !(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 2 && slot == 3)) {
+	if (slot != 1 && !(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER >= 2 && slot == 3) &&
+	    !(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 3 && slot == 5)) {
 		return IMG_MGMT_RET_RC_INVALID_SLOT;
 	}
```

Comment:
Consider the following, more generic, check for the existing secondary image slot
```
if ((slot & 1) && (slot < (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER * 2))) {
}
```